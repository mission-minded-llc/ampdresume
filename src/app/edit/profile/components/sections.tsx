import { Box, Typography } from "@mui/material";

export const InputSection = ({ children }: { children: React.ReactNode[] }) => (
  <Box
    sx={{
      display: "flex",
      flexDirection: "column",
      gap: 2,
      height: "100%",
    }}
  >
    {children}
  </Box>
);

export const GridSection = ({
  children,
  isDesktop,
}: {
  children: React.ReactNode;
  isDesktop: boolean;
}) => (
  <Box
    sx={{
      display: "grid",
      gridTemplateColumns: isDesktop ? "1fr 1fr" : "1fr",
      gap: isDesktop ? 4 : 2,
      mb: 4,
    }}
  >
    {children}
  </Box>
);

export const SectionTitle = ({ children }: { children: React.ReactNode }) => (
  <Typography
    variant="h5"
    component="h2"
    color="textPrimary"
    sx={{
      gridColumn: "1 / -1",
      borderTop: "1px solid",
      paddingTop: 2,
      marginTop: 2,
    }}
  >
    {children}
  </Typography>
);

export const FieldTitle = ({
  children,
  component = "p",
}: {
  children: React.ReactNode;
  component?: "p" | "span";
}) => (
  <Typography
    variant="h6"
    component={component}
    color="textPrimary"
    sx={{
      display: "flex",
      alignItems: "center",
      gap: "0.5em",
    }}
  >
    {children}
  </Typography>
);

export const FieldDescription = ({ children }: { children: React.ReactNode }) => (
  <Typography variant="body2" color="textSecondary">
    {children}
  </Typography>
);
