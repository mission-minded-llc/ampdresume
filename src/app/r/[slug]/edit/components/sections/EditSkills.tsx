import { Box, Container } from "@mui/material";

export const EditSkills = () => {
  return (
    <Container>
      <Box>
        <h1>Edit Skills</h1>
      </Box>
    </Container>
  );
};
