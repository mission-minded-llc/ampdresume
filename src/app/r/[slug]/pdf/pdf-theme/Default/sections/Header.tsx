import { Typography } from "@mui/material";
import { User } from "@prisma/client";

export const Header = ({ user }: { user: User }) => (
  <>
    <Typography
      sx={{ padding: 1, fontSize: 13, textAlign: "center", borderBottom: `1px solid #ccc` }}
    >
      {user?.location ? `${user.location} | ` : null}
      {user?.displayEmail ? <a href={`mailto:${user.displayEmail}`}>{user.displayEmail}</a> : null}
    </Typography>
    <Typography sx={{ mt: 1, fontSize: 26, fontWeight: "bold", textAlign: "center" }}>
      {user.name}
    </Typography>
    <Typography
      sx={{ mb: 2, mt: 1, fontSize: 13, fontWeight: "bold", textAlign: "center", letterSpacing: 0 }}
    >
      {user.title}
    </Typography>
  </>
);
