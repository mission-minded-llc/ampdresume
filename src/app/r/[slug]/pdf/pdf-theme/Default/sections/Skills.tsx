import { Box, Typography } from "@mui/material";
import { Section, SectionTitle } from "../styled";

import React from "react";
import { SkillForUserWithSkill } from "@/graphql/getSkillsForUser";
import { groupSkillsForUserByYearExperience } from "../../util";

export const Skills = ({ skillsForUser }: { skillsForUser: SkillForUserWithSkill[] }) => {
  const skillsForUserBySkill = groupSkillsForUserByYearExperience(skillsForUser);

  return (
    <Section>
      <SectionTitle>Skills</SectionTitle>
      <Box
        sx={{
          display: "grid",
          gridTemplateColumns: "auto 1fr",
          gridTemplateRows: "auto",
          width: "100%",
          gap: "5px",
          mb: 3,
        }}
      >
        {skillsForUserBySkill.map((skillsExperience) => {
          const totalYears = skillsExperience[0];
          const skillsList = skillsExperience[1];

          return (
            <React.Fragment key={`skill-group-${skillsExperience[0]}`}>
              <Box
                sx={{
                  fontWeight: "bold",
                  textAlign: "right",
                  pr: 1,
                  fontSize: 14,
                }}
              >
                {totalYears ? `${totalYears} year${parseInt(totalYears, 10) > 1 ? "s" : ""}:` : ""}
              </Box>
              <Box
                sx={{
                  display: "flex",
                  flexWrap: "wrap",
                  gap: "5px",
                  alignItems: "center",
                }}
              >
                {skillsList.map((skill, index) => (
                  <Typography key={`skill-${skill.skill.name}`} sx={{ fontSize: 14 }}>
                    {skill.skill.name}
                    {index < skillsList.length - 1 ? "," : ""}
                  </Typography>
                ))}
              </Box>
            </React.Fragment>
          );
        })}
      </Box>
    </Section>
  );
};
