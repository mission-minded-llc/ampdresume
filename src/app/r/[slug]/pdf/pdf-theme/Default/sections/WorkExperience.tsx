import { Box, Divider, Typography } from "@mui/material";
import { Section, SectionSubtitle, SectionTitle } from "../styled";

import { Company } from "@/graphql/getCompanies";
import { PositionWithSkillsForProjects } from "@/graphql/getPositionsWithSkillsForProjects";
import { formatLongDate } from "@/lib/format";

interface SectionWorkExperienceProps {
  companies: Company[];
  positionsWithSkillsForProjects: PositionWithSkillsForProjects[];
}

export const WorkExperience = ({
  companies,
  positionsWithSkillsForProjects,
}: SectionWorkExperienceProps) => {
  return (
    <Section>
      <SectionTitle>Work Experience</SectionTitle>
      {companies.map((company) => {
        const positions = positionsWithSkillsForProjects.filter(
          (position) => position.company.id === company.id && position.title !== null,
        );

        return (
          <Box key={company.id} sx={{ mb: 2 }}>
            <SectionSubtitle>{company.name}</SectionSubtitle>
            <Divider />
            {positions.map((position) => {
              return (
                <Box key={position.id} sx={{ mb: 2 }}>
                  <SectionSubtitle>
                    {position.title}
                    <Typography component="span" variant="body2">
                      {" "}
                      &mdash; {formatLongDate(position.startDate)} to{" "}
                      {position?.endDate ? formatLongDate(position.endDate) : "present"}
                    </Typography>
                  </SectionSubtitle>
                  {position.projects.map((project) => {
                    return (
                      <Typography
                        key={project.id}
                        sx={{
                          pl: 2,
                          fontSize: 14,
                          mt: 0.5,
                          "&:before": {
                            content: '"\\2022"',
                            paddingRight: "0.5em",
                            position: "absolute",
                            marginTop: "-0.1em",
                            marginLeft: "-0.5em",
                          },
                        }}
                      >
                        {project.name}{" "}
                        {project.skillsForProject.length > 0 ? (
                          <>
                            {project.skillsForProject.map((skill, index) => (
                              <Typography
                                key={skill.skillForUser.skill.name}
                                component="span"
                                sx={{
                                  fontSize: 14,
                                  color: "#666",
                                }}
                              >
                                {skill.skillForUser.skill.name}
                                {index < project.skillsForProject.length - 1 ? ", " : ""}
                              </Typography>
                            ))}
                          </>
                        ) : null}
                      </Typography>
                    );
                  })}
                </Box>
              );
            })}
          </Box>
        );
      })}
    </Section>
  );
};
