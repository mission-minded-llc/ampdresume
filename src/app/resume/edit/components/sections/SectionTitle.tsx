import { Box, Divider, Typography } from "@mui/material";

export const SectionTitle = ({ title }: { title: string }) => (
  <>
    <Divider sx={{ mb: 2 }} />
    <Box>
      <Typography variant="h5" sx={{ mb: 2 }}>
        {title}
      </Typography>
    </Box>
    <Divider sx={{ mb: 4, mt: 2 }} />
  </>
);
