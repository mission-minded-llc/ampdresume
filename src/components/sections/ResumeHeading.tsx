import { User } from "@prisma/client";
import styles from "./ResumeHeading.module.scss";

export const ResumeHeading = ({ user }: { user: User }) => {
  return (
    <h1 className={styles.userTitleHeading}>
      {user?.name}
      <br />
      <span className={styles.userTitle}>{user?.title}</span>
      <br />
      <span className={styles.userMeta}>
        {user?.displayEmail}
        <span className={styles.separator}>|</span>
        {user?.location}
      </span>
    </h1>
  );
};
