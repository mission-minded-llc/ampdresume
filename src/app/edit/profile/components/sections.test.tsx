import "@testing-library/jest-dom";

import { FieldDescription, FieldTitle, GridSection, InputSection, SectionTitle } from "./sections";

import React from "react";
import { render } from "@testing-library/react";

describe("sections components", () => {
  it("renders InputSection correctly", () => {
    const { container } = render(
      <InputSection>
        <div>Child 1</div>
        <div>Child 2</div>
      </InputSection>,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders GridSection correctly for desktop", () => {
    const { container } = render(
      <GridSection isDesktop={true}>
        <div>Child 1</div>
        <div>Child 2</div>
      </GridSection>,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders GridSection correctly for mobile", () => {
    const { container } = render(
      <GridSection isDesktop={false}>
        <div>Child 1</div>
        <div>Child 2</div>
      </GridSection>,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders SectionTitle correctly", () => {
    const { container } = render(<SectionTitle>Section Title</SectionTitle>);
    expect(container).toMatchSnapshot();
  });

  it("renders FieldTitle correctly", () => {
    const { container } = render(<FieldTitle>Field Title</FieldTitle>);
    expect(container).toMatchSnapshot();
  });

  it("renders FieldDescription correctly", () => {
    const { container } = render(<FieldDescription>Field Description</FieldDescription>);
    expect(container).toMatchSnapshot();
  });
});
