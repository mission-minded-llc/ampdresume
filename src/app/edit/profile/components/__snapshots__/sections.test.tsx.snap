// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sections components renders FieldDescription correctly 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body2 css-1a1whku-MuiTypography-root"
  >
    Field Description
  </p>
</div>
`;

exports[`sections components renders FieldTitle correctly 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-h6 css-115ey5t-MuiTypography-root"
  >
    Field Title
  </p>
</div>
`;

exports[`sections components renders GridSection correctly for desktop 1`] = `
<div>
  <div
    class="MuiBox-root css-bvujjh"
  >
    <div>
      Child 1
    </div>
    <div>
      Child 2
    </div>
  </div>
</div>
`;

exports[`sections components renders GridSection correctly for mobile 1`] = `
<div>
  <div
    class="MuiBox-root css-12dc65p"
  >
    <div>
      Child 1
    </div>
    <div>
      Child 2
    </div>
  </div>
</div>
`;

exports[`sections components renders InputSection correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-18mctc4"
  >
    <div>
      Child 1
    </div>
    <div>
      Child 2
    </div>
  </div>
</div>
`;

exports[`sections components renders SectionTitle correctly 1`] = `
<div>
  <h2
    class="MuiTypography-root MuiTypography-h5 css-xt72g8-MuiTypography-root"
  >
    Section Title
  </h2>
</div>
`;
